File: MATH2232 Numerical Methods Lab/RU/Numerical-Methods-Lab-Final-main akif/NewtonRaphson.java
package Akif;

public class NewtonRaphson {

    // A function just holding the equation
    public static double func(double x) {
        return x * x * x - 2 * x - 5;
    }

    public static double funcDerivatives(double x) {
        return 3 * x * x - 2;
    }

    public static void NewtonRaphson(double a, double EPSILON) {
        int iteration = 0;

        double h = func(a) / funcDerivatives(a); // F(Xn) / F'(Xn)

        while (Math.abs(h) >= EPSILON) {

            h = func(a) / funcDerivatives(a);
            a = a - h; // As the formula is : Xn+1 = Xn - (F(Xn) / F'(Xn))
            System.out.println("Iteration :" + (++iteration) + ", Root = " + a);
        }
    }


}

File: MATH2232 Numerical Methods Lab/RU/Numerical-Method-with-.cpp-Class-main faarjana/Newton-Raphson.cpp
#include <bits/stdc++.h>
using namespace std;

class NewtonRaphson
{
public:
    double f(double x)
    {
        return x * x * x + 4 * x * x - 1;
    }

    double derif(double x)
    {
        return 3 * x * x + 8 * x;
    }

    void newton_raphson(double x)
    {
        double h;
        for (int i = 0; i < 200; i++)
        {
            h = f(x) / derif(x);
            x = x - h;
        }
        cout << "root = " << x << endl;
    }
};

int main()
{
    double x0;
    NewtonRaphson *NF = new NewtonRaphson();
    NF->newton_raphson(0.5);
}